File: challenging/8_count_files/count_files.py
import os
import re
from pathlib import Path

def getFiles(dir):
    files = []
    for r, d, f in os.walk(dir):
        if not re.match(r'.*node_modules.*', r):
            files += [os.path.join(r, file) for file in f]
    return files

files = getFiles('../../../copilotlabs-evaluation')
print(len(files))

File: challenging/8_count_files/count_files.go
package main

import (
	"fmt"
	"os"
	"path/filepath"
)

func main() {
	var files []string
	err := filepath.Walk("../../../copilotlabs-evaluation", func(path string, info os.FileInfo, err error) error {
		if err != nil {
			return err
		}
		if !info.IsDir() {
			files = append(files, path)
		}
		return nil
	})
	if err != nil {
		panic(err)
	}
	fmt.Println(len(files))
}
